library_information_validator.py

```diff
@@ -0,0 +1,318 @@
+# load libraries
+from __future__ import print_function
+import argparse
+from openpyxl import load_workbook
+import coloredlogs
+import logging
+import re
+import couchdb
+import numbers
+import decimal
+
+
+
+
+#global variable
+WARNINGS = 0
+
+
+
+
+# Set up a logger with colored output
+logger = logging.getLogger(__name__)
+logger.propagate = False  # Otherwise the messages appeared twice
+coloredlogs.install(level='INFO', logger=logger,
+                    fmt='%(asctime)s %(levelname)s %(message)s')
+
+
+
+
+class ProjectSheet:
+#Class Attributes
+    SHEET_NAME = 'Sample information'
+    FIRST_LINE = 20  # First line where user submitted data is located
+    SAMPLE_NAME_COL = 'O'
+    A_RATIO_COL = 'R'  # A260:A280 ratio
+    CONC_COL = 'S'
+    VOL_COL = 'T'
+    RIN_COL = 'V'
+    SAMPLE_TYPE = 'O8'
+    PLATE_ID = 'M6'
+    PROJECT_NAME_USER_SHEET = 'M3'
+    sample_recommendation_sheet ='/Users/franziska.franziska/Documents/FB_NGI/bioinfo/standalone_scripts_git/standalone_scripts/Samplesheet_converter/Sample_requirements.xlsx'
+
+# Initializer / Instance attributes
+    def __init__(self, sample_info_sheet):
+        self.sample_info_sheet = sample_info_sheet
+
+# instance methods
+    def getAccessUserSheet(self):
+        wb = load_workbook(self.sample_info_sheet, read_only=True, data_only=True)
+        ws = wb[ProjectSheet.SHEET_NAME]
+        return(ws)
+
+    def projectID(self):
+        access_sample_info_sheet = self.getAccessUserSheet()
+        plate_id = access_sample_info_sheet[ProjectSheet.PLATE_ID].value
+        project_id_user = re.findall('P[0-9]+', plate_id)[0]
+        return(project_id_user)
+
+    def projectName_U(self):
+        access_sample_info_sheet = self.getAccessUserSheet()
+        project_name_user = re.split('-', access_sample_info_sheet[ProjectSheet.PROJECT_NAME_USER_SHEET].value)[0]
+        return(project_name_user)
+
+    def getSamples(self):
+        """ identifies the all rows containing a sample name, discards rows without entry.
+        Rows containing whitespace only trigger a warning and are discarded for subsequent
+        tests """
+        access_sample_info_sheet = self.getAccessUserSheet()
+        real = 1
+        cellID_withSample =list()
```

Comment:
style: spaces on both sides of the `=` operator. There's actually an official style guide for python:
https://www.python.org/dev/peps/pep-0008/